Mathlib/RingTheory/Kaehler/Basic.lean

```diff
@@ -763,6 +788,88 @@ lemma KaehlerDifferential.exact_mapBaseChange_map :
     Function.Exact (mapBaseChange R A B) (map R A B B) :=
   SetLike.ext_iff.mp (range_mapBaseChange R A B).symm
 
+/-- The map `I → B ⊗[A] B ⊗[A] Ω[A⁄R]` where `I = ker(A → B)`. -/
+@[simps]
+noncomputable
+def KaehlerDifferential.kerToTensor :
+    RingHom.ker (algebraMap A B) →ₗ[A] B ⊗[A] Ω[A⁄R] where
+  toFun x := 1 ⊗ₜ D R A x
+  map_add' x y := by simp only [AddSubmonoid.coe_add, Submodule.coe_toAddSubmonoid, map_add,
+    TensorProduct.tmul_add]
+  map_smul' r x := by simp only [SetLike.val_smul, Derivation.leibniz, RingHom.id_apply,
+    TensorProduct.tmul_smul, TensorProduct.smul_tmul', add_zero, ← Algebra.algebraMap_eq_smul_one,
+    TensorProduct.zero_tmul, smul_eq_mul, TensorProduct.tmul_add, (RingHom.mem_ker _).mp x.prop]
+
+/-- The map `I/I² → B ⊗[A] B ⊗[A] Ω[A⁄R]` where `I = ker(A → B)`. -/
+noncomputable
+def KaehlerDifferential.kerCotangentToTensor :
```

Comment:
Can you avoid tactic mode here?

Reply:
The hole is a Prop, so I think it's fine?

Reply:
I mean something like
```lean
noncomputable
def KaehlerDifferential.kerCotangentToTensor :
    (RingHom.ker (algebraMap A B)).Cotangent →ₗ[A] B ⊗[A] Ω[A⁄R] :=
  Submodule.liftQ _ (kerToTensor R A B) (iSup_le_iff.mpr (by
    simp only [Submodule.map_le_iff_le_comap, Subtype.forall]
    rintro x hx y -
    simp only [Submodule.mem_comap, LinearMap.lsmul_apply, LinearMap.mem_ker, map_smul,
      kerToTensor_apply, TensorProduct.smul_tmul', ← algebraMap_eq_smul_one,
      (RingHom.mem_ker _).mp hx, TensorProduct.zero_tmul]))
```
Or to prove the thing in a separate lemma (tbh I didn't look at what it is, so I am not sure this is reasonable).